charts/influxdb/templates/service-nodeport.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: Service
+metadata:
+{{- if .Values.service.annotations }}
+  annotations:
+{{ toYaml .Values.service.annotations | indent 4 }}
+{{- end }}
+  name: {{ template "influxdb.fullname" . }}-nodeport
+  labels:
+    app: {{ template "influxdb.fullname" . }}-nodeport
+    chart: "{{ template "influxdb.chart" . }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  type: {{ .Values.serviceNodePort.type }}
+  ports:
+  {{- if .Values.config.http.enabled }}
+  - name: api
+    port: {{ .Values.config.http.bind_address }}
+    targetPort: {{ .Values.config.http.bind_address }}
+    nodePort: {{ .Values.serviceNodePort.nodeport }}
+  {{- end }}
+  selector:
+    app: {{ template "influxdb.fullname" . }}
```

Comment:
Add Nodeport support to the influxDB service.yaml file instead of here. See other charts for examples.

Reply:
nodeport was added to service and file specific to nodeport was removed